diff -urbB bash-3.0.org/config.h.in bash-3.0/config.h.in
--- bash-3.0.org/config.h.in	2004-07-21 22:08:31.000000000 +0200
+++ bash-3.0/config.h.in	2005-10-08 19:35:26.593219000 +0200
@@ -197,7 +197,7 @@
 
 /* System paths */
 
-#define DEFAULT_MAIL_DIRECTORY "/usr/spool/mail"
+#define DEFAULT_MAIL_DIRECTORY "/var/mail"
 
 /* Characteristics of the system's header files and libraries that affect
    the compilation environment. */
Tylko w bash-3.0: config.h.in.orig
diff -urbB bash-3.0.org/config-top.h bash-3.0/config-top.h
--- bash-3.0.org/config-top.h	2003-08-05 16:36:12.000000000 +0200
+++ bash-3.0/config-top.h	2005-10-08 19:36:04.395581500 +0200
@@ -52,14 +52,14 @@
 /* The default value of the PATH variable. */
 #ifndef DEFAULT_PATH_VALUE
 #define DEFAULT_PATH_VALUE \
-  "/usr/gnu/bin:/usr/local/bin:/bin:/usr/bin:."
+  "/usr/local/bin:/bin:/usr/bin"
 #endif
 
 /* The value for PATH when invoking `command -p'.  This is only used when
    the Posix.2 confstr () function, or CS_PATH define are not present. */
 #ifndef STANDARD_UTILS_PATH
 #define STANDARD_UTILS_PATH \
-  "/bin:/usr/bin:/sbin:/usr/sbin:/etc:/usr/etc"
+  "/bin:/usr/bin:/sbin:/usr/sbin"
 #endif
 
 /* Default primary and secondary prompt strings. */
@@ -73,7 +73,7 @@
 #define KSH_COMPATIBLE_SELECT
 
 /* System-wide .bashrc file for interactive shells. */
-/* #define SYS_BASHRC "/etc/bash.bashrc" */
+#define SYS_BASHRC "/etc/bashrc"
 
 /* System-wide .bash_logout for login shells. */
 /* #define SYS_BASH_LOGOUT "/etc/bash.bash_logout" */
