--- bash-2.05b/Makefile.in	Sun Mar  2 11:45:53 2003
+++ bash-2.05b.new/Makefile.in	Sun Mar  2 12:55:29 2003
@@ -482,6 +482,8 @@
 	@echo "	  ***********************************************************"
 	@echo
 
+$(VERSPROG) $(OBJECTS) $(BUILTINS_DEP) $(LIBDEP): version.h
+
 bashbug: $(SUPPORT_SRC)bashbug.sh config.h Makefile $(VERSPROG)
 	@sed -e "s%!MACHINE!%$(Machine)%" -e "s%!OS!%$(OS)%" \
 	     -e "s%!CFLAGS!%$(CCFLAGS)%" -e "s%!CC!%$(CC)%" \
@@ -578,6 +580,7 @@
 	$(RM) $@
 	./mksyntax -o $@
 
+$(BUILTINS_LIBRARY): version.h
 $(BUILTINS_LIBRARY): $(BUILTIN_DEFS) $(BUILTIN_C_SRC) config.h ${BASHINCDIR}/memalloc.h
 	@(cd $(DEFDIR) && $(MAKE) $(MFLAGS) DEBUG=${DEBUG} libbuiltins.a ) || exit 1
 
