--- bash-2.05b/Makefile.in	Sun Mar  2 11:45:53 2003
+++ bash-2.05b.new/Makefile.in	Sun Mar  2 12:55:29 2003
@@ -482,6 +482,8 @@
 	@echo "	  ***********************************************************"
 	@echo
 
+$(VERSPROG) $(OBJECTS) $(BUILTINS_DEP) $(LIBDEP): version.h
+
 bashbug: $(SUPPORT_SRC)bashbug.sh config.h Makefile $(VERSPROG)
 	@sed -e "s%!MACHINE!%$(Machine)%" -e "s%!OS!%$(OS)%" \
 	     -e "s%!CFLAGS!%$(CCFLAGS)%" -e "s%!CC!%$(CC)%" \
