--- bash-5.3/Makefile.in~	2025-05-16 20:56:31.000000000 +0200
+++ bash-5.3/Makefile.in	2025-10-11 19:11:44.184553852 +0200
@@ -862,8 +862,8 @@ $(SUPPORT_DIR)/bashbug.sh: $(SUPPORT_DIR
 	CONFIG_FILES=$(SUPPORT_DIR)/bashbug.sh CONFIG_HEADERS= $(SHELL) ./config.status
 
 # comment out for distribution
-$(srcdir)/configure:	$(srcdir)/configure.ac $(srcdir)/aclocal.m4 $(srcdir)/config.h.in
-	cd $(srcdir) && autoconf
+#$(srcdir)/configure:	$(srcdir)/configure.ac $(srcdir)/aclocal.m4 $(srcdir)/config.h.in
+#	cd $(srcdir) && autoconf
 
 # for chet
 reconfig: force
