--- bash-3.2/jobs.c.old	2008-05-02 16:24:17.000000000 +0200
+++ bash-3.2/jobs.c	2008-05-02 16:24:34.000000000 +0200
@@ -2824,8 +2824,6 @@ start_job (job, foreground)
   else
     jobs[job]->flags &= ~J_FOREGROUND;
 
-  /* If the job is already running, then don't bother jump-starting it. */
-  if (already_running == 0)
     {
       jobs[job]->flags |= J_NOTIFIED;
       killpg (jobs[job]->pgrp, SIGCONT);
