diff -urNp -x '*.orig' bash-5.1.org/config-top.h bash-5.1/config-top.h
--- bash-5.1.org/config-top.h	2021-03-21 13:21:19.045594925 +0100
+++ bash-5.1/config-top.h	2021-03-21 13:21:19.703594925 +0100
@@ -18,6 +18,8 @@
    along with Bash.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#define NON_INTERACTIVE_LOGIN_SHELLS
+
 /* Define CONTINUE_AFTER_KILL_ERROR if you want the kill command to
    continue processing arguments after one of them fails.  This is
    what POSIX.2 specifies. */
