diff -urNp -x '*.orig' bash-5.1.org/config-top.h bash-5.1/config-top.h
--- bash-5.1.org/config-top.h	2020-11-16 16:34:26.000000000 +0100
+++ bash-5.1/config-top.h	2021-03-21 13:21:08.467594925 +0100
@@ -63,7 +63,7 @@
 /* The default value of the PATH variable. */
 #ifndef DEFAULT_PATH_VALUE
 #define DEFAULT_PATH_VALUE \
-  "/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin:."
+  "/usr/local/bin:/bin:/usr/bin"
 #endif
 
 /* If you want to unconditionally set a value for PATH in every restricted
@@ -74,7 +74,7 @@
    the Posix.2 confstr () function, or CS_PATH define are not present. */
 #ifndef STANDARD_UTILS_PATH
 #define STANDARD_UTILS_PATH \
-  "/bin:/usr/bin:/sbin:/usr/sbin:/etc:/usr/etc"
+  "/usr/local/bin:/bin:/usr/bin"
 #endif
 
 /* Default primary and secondary prompt strings. */
@@ -91,7 +91,7 @@
 #define DEFAULT_BASHRC "~/.bashrc"
 
 /* System-wide .bashrc file for interactive shells. */
-/* #define SYS_BASHRC "/etc/bash.bashrc" */
+#define SYS_BASHRC "/etc/bashrc"
 
 /* System-wide .bash_logout for login shells. */
 /* #define SYS_BASH_LOGOUT "/etc/bash.bash_logout" */
diff -urNp -x '*.orig' bash-5.1.org/config.h.in bash-5.1/config.h.in
--- bash-5.1.org/config.h.in	2020-05-19 19:54:22.000000000 +0200
+++ bash-5.1/config.h.in	2021-03-21 13:21:08.467594925 +0100
@@ -247,7 +247,7 @@
 
 /* System paths */
 
-#define DEFAULT_MAIL_DIRECTORY "/usr/spool/mail"
+#define DEFAULT_MAIL_DIRECTORY "/var/mail"
 
 /* Characteristics of the system's header files and libraries that affect
    the compilation environment. */
