--- bash-2.05a/Makefile.in	Tue Feb 12 22:55:14 2002
+++ bash-2.05a/Makefile.in.new	Tue Feb 12 23:40:24 2002
@@ -630,6 +630,7 @@
 	$(INSTALL_PROGRAM) $(INSTALLMODE) $(Program) $(DESTDIR)$(bindir)/$(Program)
 	$(INSTALL_PROGRAM) $(INSTALLMODE2) bashbug $(DESTDIR)$(bindir)/bashbug
 	-( cd $(DOCDIR) ; $(MAKE) $(MFLAGS) \
+		DESTDIR=$(DESTDIR) \
 		man1dir=$(man1dir) man1ext=$(man1ext) \
 		man3dir=$(man3dir) man3ext=$(man3ext) \
 		infodir=$(infodir) htmldir=$(htmldir) DESTDIR=$(DESTDIR) $@ )
--- bash-2.05a/doc/Makefile.in	Thu Feb  7 22:38:32 2002
+++ bash-2.05a/doc/Makefile.in.new	Tue Feb 12 23:39:40 2002
@@ -184,10 +184,10 @@
 	$(RM) Makefile
 
 installdirs:
-	-test -d $(man1dir) || $(SHELL) ${MKDIRS} $(DESTDIR)$(man1dir)
-	-test -d $(infodir) || $(SHELL) ${MKDIRS} $(DESTDIR)$(infodir)
+	-test -d $(DESTDIR)$(man1dir) || $(SHELL) ${MKDIRS} $(DESTDIR)$(man1dir)
+	-test -d $(DESTDIR)$(infodir) || $(SHELL) ${MKDIRS} $(DESTDIR)$(infodir)
 	-if test -n "$(htmldir)" ; then \
-		test -d $(htmldir) || $(SHELL) ${MKDIRS} $(DESTDIR)$(htmldir) ; \
+		test -d $(DESTDIR)$(htmldir) || $(SHELL) ${MKDIRS} $(DESTDIR)$(htmldir) ; \
 	fi
 	
 install: info installdirs
